src/kdf.rs

```diff
@@ -39,7 +41,7 @@
 macro_rules! kdf {
     ($length: ident, $($bytes: expr),+) => {
         {
-            let mut hasher = Shake256::default();
+            let mut hasher = Shake128::default();
```

Comment:
for my knowledge, is there any security harm to use Shake128 instead of Shake256?

Reply:
The security of Shake128 is [`min(d/2, 128)` bits](https://en.wikipedia.org/wiki/SHA-3#Instances), where `d` is the size of the output. This is enough for us.

The security of Share256 is `min(d/2, 256)` bits.

This is tricky because the security of AES256 or SHA256 is 128 bits.

Reply:
thank you